File: apps/spring-boot-api/src/main/java/com/app/api/user/service/UserService.java
package com.app.api.user.service;

import com.app.api.user.dto.SimpleUserDto;
import com.app.api.user.request.ChangePasswordRequest;
import com.app.api.user.request.UpdateUserRequest;
import java.util.List;

public interface UserService {
  List<SimpleUserDto> findAll();

  SimpleUserDto findByUsername(String username);

  SimpleUserDto update(UpdateUserRequest updateUserRequest);

  void changePassword(ChangePasswordRequest changePasswordRequest);
}

File: libs/angular-ui/user/src/lib/user.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { ChangePasswordRequest, SimpleUser, UpdateUserRequest } from '@bootstrapbugz/shared';

@Injectable({
  providedIn: 'root',
})
export class UserService {
  API_URL = 'localhost:8181/v1.0';

  constructor(private http: HttpClient) {}

  findAll() {
    return this.http.get<SimpleUser[]>(`${this.API_URL}/users`);
  }

  findByUsername(username: string) {
    return this.http.get<SimpleUser>(`${this.API_URL}/users/${username}`);
  }

  changePassword(changePasswordRequest: ChangePasswordRequest) {
    return this.http.put<void>(`${this.API_URL}/users/change-password`, changePasswordRequest);
  }

  updateUser(updateUser: UpdateUserRequest) {
    return this.http.put<SimpleUser>(`${this.API_URL}/users/update`, updateUser);
  }
}
